pkg/pagination/pagination.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pagination implements a series of utilities for dealing with
+// paginating lists of objects for an API.
+package pagination
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/fernet/fernet-go"
+)
+
+// ErrInvalidToken is returned when a token fails to Unmarshal because it was
+// invalid or expired.
+var ErrInvalidToken = errors.New("invalid or expired pagination token")
+
+// Key represents the key used to cryptographically secure the token
+// being used to keep track of pages.
+type Key string
+
+// MustGenerateNewKey generates a new pagination key.
+//
+// This function will panic if your system does not have enough entropy to
+// produce the required randomness.
+func MustGenerateNewKey() Key {
+	var key fernet.Key
+	if err := key.Generate(); err != nil {
+		panic("pagination: failed to read enough randomness to generate a new key")
```

Comment:
I understand that by current implementation, it'll only return error because of insufficient randomness. But will this error be the only error that key.Generate function will generate in the future?